packages/compiler/test/server/watched-file-changed.test.ts

```diff
@@ -0,0 +1,112 @@
+import { deepStrictEqual } from "assert";
```

Comment:
please also add test that if only add or only remove a file won't trigger anything.